package/MDAnalysis/analysis/base.py

```diff
@@ -24,61 +24,116 @@
 """
 from six.moves import range
 
-import numpy as np
 import logging
 from MDAnalysis.lib.log import ProgressMeter
 
 logger = logging.getLogger(__name__)
 
 
 class AnalysisBase(object):
-    """Base class for defining multi frame analysis
-
-    The analysis base class is designed as a template for creating
-    multiframe analysis.
-
-    The class implements the following methods:
-
-    _setup_frames(trajectory, start=None, stop=None, step=None)
-      Pass a Reader object and define the desired iteration pattern
-      through the trajectory
-
-    run
-      The user facing run method.  Calls the analysis methods
-      defined below
-
-    Your analysis can implement the following methods, which are
-    called from run:
-
-    _prepare
-      Called before iteration on the trajectory has begun.
-      Data structures can be set up at this time, however most
-      error checking should be done in the __init__
-
-    _single_frame
-      Called after the trajectory is moved onto each new frame.
-
-    _conclude
-      Called once iteration on the trajectory is finished.
-      Apply normalisation and averaging to results here.
+    """Base class for defining multi frame analysis, it is designed as a
+    template for creating multiframe analysis. This class will automatically
+    take care of setting up the trajectory reader for iterating and offers to
+    show a progress meter.
+
+    To define a new Analysis, `AnalysisBase` needs to be subclassed
+    `_single_frame` must be defined. It is also possible to define
+    `_prepare` and `_conclude` for pre and post processing. See the example
+    below.
+
+    .. code-block:: python
```

Comment:
This code block is now *the* definition of the `AnalysisBase` API. There's a danger that this will be overlooked. Furthermore, one cannot easily link to it.

There should be some text at the module level similar to the one that used to be here and formally describe the API.